### DISCUSSION TITLE
Interesting observation: Sourcing .zprofile in .zshenv breaks running VS Code from within bottle

### DISCUSSION THREAD

**Discussion by trallnag** [arkane-systems/genie#205] [Show and tell]
If I add the following block to `~/.zshenv` it breaks running / starting VS Code with `code` from within genie.

```
# Ensure that a non-login, non-interactive shell has a defined environment.
if [[ ( "$SHLVL" -eq 1 && ! -o LOGIN ) && -s "${ZDOTDIR:-$HOME}/.zprofile" ]]; then
  source "${ZDOTDIR:-$HOME}/.zprofile"
fi
```

Can somebody who uses Zsh try to reproduce this? I got the block from either Oh My Zsh or Prezto

**Comment by PavelSosin-320** [comment]
@trallnag I suppose that you intend to use systemd-genie to managing systemd services. If so, systemd genie's genie -l starts a new pam/logind session, not a shell session. It's the fact that systemd units don't inherit session environment automatically - every service can run in its own environment passed to the unit using Environment directive or environment file. This is the systemd concept. At any point in your session you can use either systemctl status and systemctl --user status to see your current shell and the current environment. . genie's init file allows passing certain environment variables to the session but they are not taken from the session environment.
Anyway, the real service hierarchy and environment are very dependant on the distro and your service configurations that are documented separately. Systemd service manager triggered by genie and X-terminal session mechanism included into any Linux  desktop distros manage start login shell as a service with environment different from user-to-user and for root user separately.
The shortest explanation that I ever saw is in [Systemd User WiKi](https://wiki.archlinux.org/title/systemd/User) in the "Basic setup" paragraph.
